docs/root/faq/edge.rst

```diff
@@ -0,0 +1,85 @@
+.. _faq_edge:
+
+Can I use Envoy as an edge proxy?
+=================================
+
+Envoy is a production-ready edge proxy, and performs well in that role, however,
+the default settings are tailored for the service mesh use case, and some values
+need to be adjusted when using Envoy as an edge proxy.
+
+TCP proxies should configure: :ref:`overload_manager <config_overload_manager>`,
+:ref:`listener buffer limits <envoy_api_field_Listener.per_connection_buffer_limit_bytes>` to 32,768 bytes
```

Comment:
nit: I would just use units of KiB and MiB, so 32KiB, 1MiB, etc. Here and below.

Reply:
How is this value derived? We're using 16MiB in our configuration, for example, to allow more buffering of larger requests. Has Envoy been built in a particular way that 32KiB per-conn buffers are optimal for the edge proxy use-case?

Reply:
This is not really envoy specific. We have not benchmarked envoy to determine what buffer size would be optimal to amortize connections that are CPU hogs (i.e. connections that use a lot of small frames and cause CPU burn in the framer) most effectively. Not sure how practical such benchmarking could be. This value is used in google GFEs and we took it as a guidepost for something that have been tuned over the years.
There is no reason to make it this setting larger than the net.ipv4.tcp_rmem default (86k on by default on debian kernels).  It is highly unlikely that you'd ever get a read() more than that in edge use cases before EGAIN is returned. After that it is just wasted memory for each listener thread.